src/tests/multihost/alltests/test_misc_proxy.py

```diff
@@ -0,0 +1,240 @@
+""" Automation of proxy provider suite
+
+:requirement: IDM-SSSD-REQ : Proxy Provider
+:casecomponent: sssd
+:subsystemteam: sst_idm_sssd
+:upstream: yes
+"""
+from __future__ import print_function
+import pytest
+import subprocess
+import ldap
+from sssd.testlib.common.utils import sssdTools, SSHClient, LdapOperations
+
+
+def execute_cmd(multihost, command):
+    """ Execute command on client """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+@pytest.mark.usefixtures('setup_sssd_krb',
+                         'create_posix_usersgroups',
+                         'sssdproxyldap',
+                         'install_nslcd',
+                         'misc_test',
+                         'netgroups')
+@pytest.mark.tier1_3
+class TestProxyMisc(object):
+    """
+    This is test case class for proxy provider suite
+    """
+    def test_bz1036758(self, multihost, backupsssdconf):
+        """
+        :title: SSSD: Allow for custom attributes in
+         RDN when using id_provider = proxy
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1036758
+        :id: 10eb49a4-b252-11ec-87ea-845cf3eff344
+        :steps:
+          1. SSSD to allow for custom attributes
+        :expectedresults:
+          1. Should succeed
+        """
+        tools = sssdTools(multihost.client[0])
```

Comment:
Hi,

sssd.conf is wrong for this test. The bash version is using_
```
    id_provider = proxy
    proxy_lib_name = files
    auth_provider = ldap
```
while here
```
id_provider = ldap
auth_provider = proxy
proxy_pam_target = sssdproxyldap
```
(main config from `setup_sssd_krb` and some tuning in `misc_test`) is used.

To verify the original bugzilla issue it is important that `auth_provider = ldap` while `id_provider` can be anything else but not `ldap`.

Since the next test is just modifying single options of sssd.conf they might be wrong as well. The next one `test_bz785902` only changes `use_fully_qualified_names` while the next bash test has
```
id_provider = proxy
auth_provider = proxy
proxy_lib_name = ldap
proxy_pam_target = sssdproxyldap
```
so sssd.conf is still different. I hven't checked to following tests.

So I would suggest to work with a complete sssd.conf similar as in the bash tests and make sure the correct one is used by individual fixtures or by explicitly setting it in the test itself.

bye,
Sumit

Reply:
changed

Reply:
changed done by explicitly setting sssd.conf in the test itself.